crates/rpc/rpc-eth-api/src/helpers/error.rs

```diff
@@ -0,0 +1,88 @@
+//! Helper traits to wrap generic l1 errors, in network specific error type configured in
+//! [`EthApiTypes`](crate::EthApiTypes).
+
+use reth_rpc_eth_types::EthApiError;
+use revm_primitives::EVMError;
+
+/// Helper trait to wrap core [`EthApiError`].
+pub trait FromEthApiError: From<EthApiError> {
+    /// Converts from error via [`EthApiError`].
+    fn from_err<E>(err: E) -> Self
+    where
+        EthApiError: From<E>;
+}
+
+impl<T> FromEthApiError for T
+where
+    T: From<EthApiError>,
+{
+    fn from_err<E>(err: E) -> Self
+    where
+        EthApiError: From<E>,
+    {
+        T::from(EthApiError::from(err))
+    }
+}
```

Comment:
ah this is required to do `error -> ethapierror -> Types::Error`

Reply:
yeah, abstracts away a lot of noise in the default eth api impl